provider/datagen/src/lib.rs

```diff
@@ -256,6 +256,8 @@ pub enum Out {
         insert_feature_gates: bool,
         /// Whether to use separate crates to name types instead of the `icu` metacrate
         use_separate_crates: bool,
+        /// Whether to overwrite existing data
+        overwrite: bool,
```

Comment:
Issue: As you mentioned, this is a breaking API change in `icu4x-datagen`. It will actually affect people if they are using the datagen API; _maybe_ not enough people are using it yet.

Reply:
Well the thing is marked `#[non_exhaustive]`, the variants just aren't...

Reply:
Currently datagen overwrites by default which is quite something. We could treat that as a bug and fail if the directory exists without offering the option to overwrite.

Reply:
Yeah, which means it's a breaking change

Reply:
Ok I'll make it fail instead of rewrite.

Reply:
If it currently overwrites, it should most likely continue to overwrite